modules/libcom/RTEMS/epicsNtp.c

```diff
@@ -51,24 +51,29 @@ int epicsNtpGetTime(char *ntpIp, struct timespec *now)
 
   // Call up the server using its IP address and port number.
   if ( connect( sockfd, ( struct sockaddr * ) &serv_addr, sizeof( serv_addr) ) < 0 ) {
+    close( sockfd );
     perror( "epicsNtpGetTime" );
     return -1;
   }
 
   // Send it the NTP packet it wants. If n == -1, it failed.
   n = write( sockfd, ( char* ) &packet, sizeof( ntp_packet ) );
   if ( n < 0 ) {
+    close( sockfd );
     perror( "epicsNtpGetTime" );
     return -1;
   }
 
 // Wait and receive the packet back from the server. If n == -1, it failed.
   n = read( sockfd, ( char* ) &packet, sizeof( ntp_packet ) );
   if ( n < 0 ) {
+    close( sockfd );
     perror( "epicsNtpGetTime" );
     return -1;
```

Comment:
`close()` may clobber the `errno` which `perror()` sees.  Simplest to `close()` after printing the error.  Also, it would be nice if the three calls to `perror()` printed different messages so that they could be distinguished.

Reply:
Thanks for the review. I can make those changes.

A couple of things to note:

1. I am planning to make further changes to this module so it can be used with LibBSD and the Legacy networking stack. The NTP provider for RTEMS in EPICS uses code that is only part of the Legacy stack and that code will not be moved to LibBSD. The code in EPICS calls [rtems_bsdnet_get_ntp()](https://git.rtems.org/rtems-net-legacy/tree/lib/rtems_bsdnet_ntp.c#n154) and I think this code is best moved into this file and the support be self contained. This change could be merged into that work?
2. The change does nothing at the moment because the code is disabled. The patch [ntp-provider-with-rtems-posix](https://github.com/epics-base/epics-base/compare/7.0...kiwichris:epics-base:ntp-provider-with-rtems-posix) enables it but this change will not build with LibBSD.

Reply:
An update from me on this PR. 

We have moved to using `ntpd` in RTEMS for Gemini and not this code or this option.